libs/staging-provider/src/AbstractProviderInitializer.php

```diff
@@ -5,42 +5,108 @@
 namespace Keboola\StagingProvider;
 
 use Keboola\InputMapping\Staging\AbstractStrategyFactory;
+use Keboola\InputMapping\Staging\ProviderInterface;
 use Keboola\InputMapping\Staging\Scope;
-use Keboola\StagingProvider\Provider\LocalStagingProvider;
-use Keboola\StagingProvider\Staging\LocalStaging;
-use Keboola\StagingProvider\Staging\Workspace\WorkspaceStagingInterface;
-use Keboola\StagingProvider\WorkspaceProviderFactory\WorkspaceProviderFactoryInterface;
 
 abstract class AbstractProviderInitializer
 {
     public function __construct(
         private readonly AbstractStrategyFactory $stagingFactory,
-        private readonly WorkspaceProviderFactoryInterface $workspaceProviderFactory,
-        private readonly string $dataDirectory,
+        private readonly ProviderInterface $workspaceStagingProvider,
+        private readonly ProviderInterface $localStagingProvider,
```

Comment:
Tohle vypada nebezpecne, jako 3. argument by si mel urcit primo vyzadovat `LocalStagingProvider`. A pak se rovnou nabizi otazka, jestli nove `LocalStagingProvider` potrebuje implementovat `ProviderInterface`, kdyz nektere metody pak na pevno hazou vyjimky. Kdyby nebylo potreba, pak by se dal jeste `ProviderInterface` prejmenovat na `WorkspaceProviedrInterface`.

Reply:
Zavedeni `WorkspaceProviderInterface` ti pak navic umozni v `docker-bundle` misto checku na abstract class checkovat hezky interface (`!$stagingProvider instanceof AbstractWorkspaceProvider` -> !$stagingProvider instanceof `WorkspaceProviderInterface`)

Reply:
> A pak se rovnou nabizi otazka, jestli nove LocalStagingProvider potrebuje implementovat ProviderInterface
Jo, potřebuje https://github.com/keboola/input-mapping/blob/main/src/Table/Strategy/AbstractStrategy.php#L31

>  misto checku na abstract class checkovat hezky interface
No ne úplně, protože tam právě potřebuju rozlišit oba případy
https://github.com/keboola/docker-bundle/pull/762/files#diff-572fc133f255533867b23e1f8cb4e9e73d68b726b2d5254ddd19904ff1eb09bbL317
Naopak bych možná měl tendenci tam v docker bundlu napsat 
místo `if (!$stagingProvider instanceof AbstractWorkspaceProvider) { `
rovnou
`if (!$stagingProvider instanceof NewWorkspaceStagingProvider || !$stagingProvider instanceof ExistingWorkspaceStagingProvider) {`
protože to je vlastně věc, kterou nechceš automaticky dělat pro každou další implementing class (ne, že bych měl nějakou v plánu) , ale zvážit pro co konkrétně se má dělat.

Reply:
Nevím no jestli dělat ten WorkspaceProviedrInterface, může se, ale není reálně na nic potřeba, takže moc nevidím důvod

Reply:
Je to jak říkáš, tam se počítalo s tím, že metadata budou i v ABS (databricks), ale nezdá se, že by to teď bylo na pořadu dne. Už jsem to tam narval napevno